server/storage.lua

```diff
@@ -164,7 +209,7 @@ function FetchPlayerEntity(citizenId)
         charinfo = json.decode(player.charinfo),
         job = player.job and json.decode(player.job),
         gang = player.gang and json.decode(player.gang),
-        position = json.decode(player.position),
+        position = vec4(position.x or QBCore.Config.DefaultSpawn.x, position.y or QBCore.Config.DefaultSpawn.y, position.z or QBCore.Config.DefaultSpawn.z, position.w or QBCore.Config.DefaultSpawn.w),
```

Comment:
This code implies that individual axis of the vec4 could fail and revert to default, but I think the desired behavior would be just to let heading default if position is stored as a vec3, otherwise should default to default spawn vec4.

Reply:
I just added the code there because json.decode doesn't return a vector and it needs to be a vector. I added fallbacks just in case anything went wrong, I don't know if it will but if anyone decides to edit it, we have a fallback.

Reply:
I'm not sure you understood what I wrote. I don't think it will occur, but the code implies that an x value could be stored correctly, but we'd need a y fallback. So the player would be on their last logged out x axis, but a default y axis.

Reply:
Oh I get it, I'll make the adjustment